### PR TITLE
GitHub E2E tests workflow: Archive results on failure

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This code has been authored by @bsessions85 and adds artifact gathering upon test failure for our E2E tests. This should allow us to debug failing tests more easily, especially when they only fail on GiHub.

Thanks to @bsessions85 for writing the code, and to @iamchughmayank for bringing up the idea.

## How has this been tested?
We got our artifacts file (https://github.com/Parsely/wp-parsely/actions/runs/6099498942?pr=1850), so we know the code works.

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -45,3 +45,10 @@ jobs:
       - name: Stop WordPress environment
         run: |
           npm run dev:stop
+
+      - name: Archive e2e results
+        uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: test-results
+          path: artifacts
```
Comment (mehmoodak) [inline .github/workflows/e2e-tests.yml:54]: Can you please explain a bit where are we storing the results and how can we view it if needed?

Comment (acicovic) [reply .github/workflows/e2e-tests.yml:54]: Yes! When E2E tests fail, and you go to the particular action (https://github.com/Parsely/wp-parsely/actions/runs/6099498942?pr=1850 for this PR), you get an Artifacts file that you can download and use for help in debugging.

Comment (acicovic) [reply .github/workflows/e2e-tests.yml:54]: Added this in a "How has this been tested?" section in the PR's description.

**Review by mehmoodak — APPROVED** [review]
Approving the PR but please also take Mo review on it as I haven't much expertise related to this.

**Comment by acicovic** [comment]
> Approving the PR but please also take Mo review on it as I haven't much expertise related to this.

Thanks @mehmoodak! Given the code is simple, woks, and doesn't touch real plugin code, I think we're safe to merge it without involving Mo so we don't put more load on him.

**Comment by mehmoodak** [comment]
No concerns from my side. Your call 🙂

**Pull request merged**